src/components/taskDetails/TaskUpdateModal.tsx

```diff
@@ -26,9 +26,22 @@ function TaskUpdateModal({
     editedTaskDetails,
     onUpdateSuccess,
 }: Props) {
+    const router = useRouter();
+    const { dev } = router.query;
+
     return (
         <Modal isOpen={isOpen} toggle={() => setIsOpen(false)}>
             <div className={styles.taskUpdateModal}>
+                {dev === 'true' && (
+                    <button
+                        className={styles.closeButton}
+                        onClick={() => setIsOpen(false)}
+                        data-testid="task-close-modal"
```

Comment:
Can we please rename this `data-testid` to something more descriptive  and also please use same `data-testid` in the tests
```suggestion
                        data-testid="task-update-modal-close-button"
```